ci/README.md

```diff
@@ -0,0 +1,210 @@
+This is good for testing compatibility with e.g. LTS distributions, but becomes problematic when we want to test more cutting-edge features, such as newer C++ standards (C++17, C++20, etc...).
+This is the reason why we have this action: we have full control over the toolchain, and it's cross platform.
+
+### Codecov
+
+**Config**: [.codecov.yml](https://github.com/dlang/dmd/blob/master/.codecov.yml)
+
+**Checks**:
+- codecov/patch
+
+Codecov checks code coverage, meaning that the lines of code you modified are executed at least once in the test suite.
+While there are currently no hard constraints on coverage for a Pull Request to pass the check, it does emit warnings for modified lines that are not covered into the "Files Changed" tab.
+These should be taken seriously, since untested code is likely to introduce bugs, and can easily be accidentally broken by future changes.
+
+### CirrusCI
+
+**Config**: [.cirrus.yml](https://github.com/dlang/dmd/blob/master/.cirrus.yml)
+
+**Checks**:
+- FreeBSD 11.4 x64, DMD (bootstrap)
+- FreeBSD 12.2 x64, DMD (coverage)
+- FreeBSD 12.2 x64, DMD (latest)
+- Ubuntu 18.04 x64, DMD (bootstrap)
+- Ubuntu 18.04 x64, DMD (latest)
+- Ubuntu 18.04 x64, GDC
+- Ubuntu 18.04 x64, LDC
+- Ubuntu 18.04 x86, DMD (bootstrap)
+- Ubuntu 18.04 x86, DMD (coverage)
+- Ubuntu 18.04 x86, DMD (latest)
+- macOS 10.15 x64, DMD (bootstrap)
+- macOS 11.x x64, DMD (coverage)
+- macOS 11.x x64, DMD (latest)
+- macOS 12.x x64, DMD (coverage)
+- macOS 12.x x64, DMD (latest)
+
+Cirrus tests DMD on Posix platforms.
+
+Since DMD is written in D, a "Host D Compiler" is needed to compile it.
+Various host compilers are tested, such as GDC, LDC, the latest DMD, and an older verison of DMD (bootstrap).
+Note that the GDC and LDC targets only test a different host compiler, they still compile the DMD compiler only.
+
+Sometimes the macOS VMs get corrupted.
+When that happens, you can file an issue on Cirrus' issue tracker, [like this one](https://github.com/cirruslabs/macos-image-templates/issues/43).
+
+### pre-commit
+
+**Config**: [.pre-commit-config.yaml](https://github.com/dlang/dmd/blob/master/.pre-commit-config.yaml)
+
+**Checks**:
+- pre_commit
+
+pre-commit checks for redundant white space, such as trailing whitespace and newlines.
+
+If you want to automatically make your changes comply with the check, you can download `pre-commit` via your package manager, or with  `pip`.
+See https://pre-commit.com/.
+
+Run `pre-commit install` inside the root of the repository,
+This will add a hook into `.git/hooks/pre-commit`.
+Every time you do a commit it will do pre-commit checks.
+
+If you already have a custom `pre-commit` hook, you can modify the bash script to run `pre-commit` manually with `pre-commit run`.
+pre-commit is designed to run only on diffed files so it will run pretty quickly, but if you want to for it to run on all files you can do `pre-commit run --all-files`.
```

Comment:
```suggestion
pre-commit is designed to run only on diffed files so it will run pretty quickly, but if you want it to run on all files you can do `pre-commit run --all-files`.
```